### PR TITLE
Yes no prompt

### PR DESCRIPTION
**Pull request by PikalaxALT**
Also fixes a longstanding bug forcing all source files to be rebuilt with each make call.

Builds on #284

### CODE REVIEW

**Comment by adrienntindall** [comment]
Address conflicting files

File: include/alph_puzzle.h [hunk 1]
```diff
@@ -18,6 +18,16 @@ typedef enum AlphPuzzles {
     ALPH_PUZZLE_MAX
 } AlphPuzzles;
 
+typedef struct UnkAlphSub_10 {
+    void *unk_00; // unused
```
Comment (adrienntindall) [inline include/alph_puzzle.h:22]: unk -> unused/filler/something else in struct field names

File: include/filesystem_files_def.h [hunk 2]
```diff
@@ -157,7 +157,7 @@ typedef enum NarcId
     NARC_a_1_4_9                                            = 149,
     NARC_a_1_5_0                                            = 150,
     NARC_a_1_5_1                                            = 151,
-    NARC_a_1_5_2                                            = 152,
+    NARC_system_touch_subwindow                                            = 152,
```
Comment (adrienntindall) [inline include/filesystem_files_def.h:160]: fix spacing

File: include/unk_0201F79C.h [hunk 3]
```diff
@@ -4,11 +4,11 @@
 #include "unk_0200E320.h"
 
 typedef struct UnkStruct_0201F79C_Sub UNK_0201F79C_Sub;
-typedef struct UnkStruct_0201F79C UNK_0201F79C;
+typedef struct UnkStruct_0201F79C SysTaskQueue;
```
Comment (adrienntindall) [inline include/unk_0201F79C.h:7]: UnkStruct_0201F79C -> SysTaskQueue

File: include/unk_0201F79C.h [hunk 4]
```diff
@@ -4,11 +4,11 @@
...
 
-void sub_0201F880(UNK_0201F79C *a0);
+void sub_0201F880(SysTaskQueue *a0);
```
Comment (adrienntindall) [inline include/unk_0201F79C.h:9]: document a0

File: include/unk_02026C44.h [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+#define POKEHEARTGOLD_UNK_02026C44_H
+
+#include "heap.h"
+#include "touchscreen.h"
+
+typedef struct TouchHitboxController TouchHitboxController;
+
+TouchHitboxController *TouchHitboxController_Create(const TouchscreenHitbox *hitboxes, u32 num_templates, void (*callback)(u32, u32, void *), void *arg, HeapID heapId);
+BOOL TouchHitboxController_IsTriggered(TouchHitboxController *a0);
+void TouchHitboxController_Destroy(TouchHitboxController *a0);
```
Comment (adrienntindall) [inline include/unk_02026C44.h:11]: document a0

File: src/alph_puzzle.c [hunk 6]
```diff
@@ -98,10 +88,10 @@ typedef struct AlphPuzzleTile {
 
 typedef struct AlphPuzzleData {
     HeapID heapId;
-    int unk4;
+    int menuIgnoreTouchFlag;
     int unkState;
     u16 subState;
-    u16 subStateTimer;
+    u16 sceneTImer;
```
Comment (adrienntindall) [inline src/alph_puzzle.c:94]: sceneTImer -> sceneTimer

File: src/alph_puzzle.c [hunk 7]
```diff
@@ -859,9 +849,9 @@ static int AlphPuzzleMainSeq_RotateTile_impl(AlphPuzzleData *data) {
         data->subState++;
         break;
     case 1:
-        u16 temp = data->subStateTimer++;
+        u16 temp = data->sceneTImer++;
```
Comment (adrienntindall) [inline src/alph_puzzle.c:852]: data->sceneTImer -> data->sceneTimer (I missed the struct somewhere so update it there too)

File: src/alph_puzzle.c [hunk 8]
```diff
@@ -1247,31 +1237,31 @@ static void AlphPuzzle_Quit_CreateYesNoPrompt(AlphPuzzleData *data) {
...
-    unkStruct.unk8 = 31;
-    unkStruct.unkC = 6;
-    unkStruct.unk4 = 0;
-    unkStruct.unk10 = 25;
-    unkStruct.unk11 = 10;
-    unkStruct.unk12_0 = data->unk4;
-    unkStruct.unk12_4 = 1;
-    unkStruct.unk13 = 0;
+    unkStruct.tileStart = 31;
+    unkStruct.plttSlot = 6;
```
Comment (adrienntindall) [inline src/alph_puzzle.c:1241]: I kinda hate ``pltt`` instead of ``palette`` but it's not the biggest of issues

File: src/credits/credits.c [hunk 9]
```diff
@@ -195,11 +195,11 @@ static void HandleSceneFlip(CreditsAppWork *work);
 static void HandleCutscenes(CreditsAppWork *work);
 static void ActivateSprite(Sprite *a0);
 static void ov76_021E68C8(CreditsAppWork *work);
-static void FlipScreensCB(int a0, ScreenFlipWork *a1, int a2);
+static void FlipScreensCB(SysTask *a0, void *a1);
```
Comment (adrienntindall) [inline src/credits/credits.c:198]: document a0

File: src/credits/credits.c [hunk 10]
```diff
@@ -195,11 +195,11 @@ static void HandleSceneFlip(CreditsAppWork *work);
...
 static void DisplayWindow(CreditsAppWork *work);
 static void ov76_021E6944(PageDisplayWork *a0, BgConfig *a1, BOOL a2);
 static BOOL PageWindowRendering(PageDisplayWork *a0);
-static void TogglePageDisplayCB(int a0, PageDisplayWork *a1, int a2);
+static void TogglePageDisplayCB(SysTask *a0, void *a1);
```
Comment (adrienntindall) [inline src/credits/credits.c:202]: document a0

File: src/credits/credits.c [hunk 11]
```diff
@@ -654,8 +654,8 @@ static void InitCutsceneSpriteResources(u8 idx, CreditsAppWork *work, u8 sprtRes
 
 static void SetPageSysTasks(CreditsAppWork *work) {
     PageWork *ptr = &work->pageWork;
-    ptr->pageDisplayWork.sysTask = sub_0200E33C((UnkSysTaskFunc)TogglePageDisplayCB, &ptr->pageDisplayWork, 0);
-    work->pageWork.scrFlipWork.sysTask = sub_0200E33C((UnkSysTaskFunc)FlipScreensCB, &work->pageWork.scrFlipWork, 1);
+    ptr->pageDisplayWork.sysTask = sub_0200E33C(TogglePageDisplayCB, &ptr->pageDisplayWork, 0);
+    work->pageWork.scrFlipWork.sysTask = sub_0200E33C(FlipScreensCB, &work->pageWork.scrFlipWork, 1);
```
Comment (adrienntindall) [inline src/credits/credits.c:658]: phase out work terminology in these fields

File: src/credits/credits.c [hunk 12]
```diff
@@ -825,7 +825,8 @@ static void ov76_021E68C8(CreditsAppWork *work) {
     ptr->count++;
 }
 
-static void FlipScreensCB(int a0, ScreenFlipWork *a1, int a2) {
+static void FlipScreensCB(SysTask *a0, void *a1v) {
+    ScreenFlipWork *a1 = (ScreenFlipWork *)a1v;
```
Comment (adrienntindall) [inline src/credits/credits.c:829]: document a1, phase out work terminology

File: src/credits/credits.c [hunk 13]
```diff
@@ -867,7 +868,8 @@ static BOOL PageWindowRendering(PageDisplayWork *a0) {
     return a0->rendering;
 }
 
-static void TogglePageDisplayCB(int a0, PageDisplayWork *a1, int a2) {
+static void TogglePageDisplayCB(SysTask *a0, void *a1v) {
+    PageDisplayWork *a1 = (PageDisplayWork *)a1v;
```
Comment (adrienntindall) [inline src/credits/credits.c:872]: Document a1, phase out work terminology

File: src/touch_save_app.c [hunk 14]
```diff
@@ -361,20 +361,22 @@ static BOOL TouchSaveApp_GetSaveConfirmation(TouchSaveAppData *data) {
...
                 data->state = TOUCHSAVEAPP_STATE_PRINT_SAVING_MESSAGE;
             }
             break;
-        case 2: // No
-            ov30_0225DC00(&data->fieldSystem->unk_10C, sub_020169C0(data->yesNoPrompt));
+        case YESNORESPONSE_NO:
+            ov30_0225DC00(&data->fieldSystem->unk_10C, YesNoPrompt_IsInTouchMode(data->yesNoPrompt));
             YesNoPrompt_Destroy(data->yesNoPrompt);
             return TRUE;
+        default:  // clang(-Wswitch)
```
Comment (adrienntindall) [inline src/touch_save_app.c:378]: what's the clang comment here for?

File: src/touch_save_app.c [hunk 15]
```diff
@@ -397,16 +399,18 @@ static BOOL TouchSaveApp_GetOverwriteConfirmation(TouchSaveAppData *data) {
 }
 
 static BOOL TouchSaveApp_HandleOverwriteConfirmation(TouchSaveAppData *data) {
-    switch (sub_02016748(data->yesNoPrompt)) {
-        case 1: // Yes
-            ov30_0225DC00(&data->fieldSystem->unk_10C, sub_020169C0(data->yesNoPrompt));
+    switch (YesNoPrompt_HandleInputForSave(data->yesNoPrompt)) {
+        case YESNORESPONSE_YES: // Yes
```
Comment (adrienntindall) [inline src/touch_save_app.c:403]: remove redundant comment

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: include/alph_puzzle.h (hunk 1 above)
Comment (PikalaxALT) [reply include/alph_puzzle.h:22]: Deferring to #290

File: include/unk_0201F79C.h (hunk 3 above)
Comment (PikalaxALT) [reply include/unk_0201F79C.h:7]: Deferred to #286

File: include/unk_0201F79C.h (hunk 4 above)
Comment (PikalaxALT) [reply include/unk_0201F79C.h:9]: Deferred to #286

File: src/credits/credits.c (hunk 11 above)
Comment (PikalaxALT) [reply src/credits/credits.c:658]: out of scope for this PR

File: src/credits/credits.c (hunk 12 above)
Comment (PikalaxALT) [reply src/credits/credits.c:829]: Phasing out work terminology is out of scope

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**